### ISSUE TITLE
Collection.from_file parses assets incorrectly

### ISSUE THREAD

**Issue by TomAugspurger** [stac-utils/pystac#378]
I think that `Collection.from_file` parses asset objects incorrectly. Given the catalog

```json
{
    "id": "id",
    "stac_version": "1.0.0",
    "type": "Collection",
    "description": "description",
    "license": "license",
    "extent": {
        "temporal": [null, null],
        "spatial": [[null, null]]
    },
    "links": {},
    "assets": {
        "a": {
            "href": "href",
            "title": "tile",
            "description": "description"
        }
    }
}
```

`pystac.Collection.from_file` gives

```python
In [12]: pystac.Collection.from_file("cat.json").assets['a'].href
Out[12]: {'href': 'href', 'title': 'tile', 'description': 'description'}
```

It looks like the whole object is being set for the `href`, rather than just the `href`.

**Comment by TomAugspurger** [comment]
Maybe just missing a `**` at https://github.com/stac-utils/pystac/blob/e90f80954163292fa9804b5a8e3cb6661ed918a7/pystac/collection.py#L718? I'll try that.

**Comment by gadomski** [comment]
Related to https://github.com/stac-utils/pystac/pull/373.

**Comment by TomAugspurger** [comment]
Ah missed that. Let's use your PR then, since from_dict is the proper way to do it?

**Comment by duckontheweb** [comment]
Closed by #373
